code/core/src/level/elements/room/Tile.java

```diff
@@ -98,9 +98,9 @@ public void addConnection(Tile to) {
      */
     public Direction[] directionTo(Tile goal) {
         List<Direction> directions = new ArrayList<>();
-        if (globalPosition.x < goal.getGlobalPosition().x) {
+        if (globalPosition.x < goal.getCoordinate().x) {
             directions.add(Direction.E);
-        } else if (globalPosition.x > goal.getGlobalPosition().x) {
+        } else if (globalPosition.x > goal.getCoordinate().x) {
             directions.add(Direction.W);
         } else if (globalPosition.y < goal.globalPosition.y) {
```

Comment:
```suggestion
        } else if (globalPosition.y < goal.getGlobalPosition().y) {
```
Hier wird noch direkt auf das Attribut zugegriffen und nicht wie zuvor über die Methode. selbe spiel in Zeile 107